heudiconv/utils.py

```diff
@@ -105,7 +105,14 @@ def dec(obj):
 def anonymize_sid(sid, anon_sid_cmd):
     from subprocess import check_output
     cmd = [anon_sid_cmd, sid]
-    return check_output(cmd).strip()
+    shell_out = check_output(cmd).strip()
+    if isinstance(shell_out, bytes):
+        anon_sid = shell_out.decode()
+    elif isinstance(shell_out, str):
+        anon_sid = shell_out
```

Comment:
we still support python2, there `bytes is str` and barely anywhere code has `unicode` in mind for python2.  I wonder if then we would start encountering unicode where we don't expect.
It would probable be worth adding a test for using this functionality, which we do not have seems to me at all.
Would you mind for now just to extend `heudiconv/tests/test_regression.py::test_conversion` with the additional option for heudiconv invocation providing some basic anonimization command?  it would help to make sure that this code works across pythons.
If brave enough, in subsequent commit try to make that function return some proper utf-8 encoded sid, and see if things still work ;)

Reply:
Added a pytest with the `--anon-cmd` flag and a variable passed from bash. Also fixed `anonymize_sid()` so that it works with both py2 and 3 (tested with 2.7.15 and 3.6.6).